tests/test_pagination.py

```diff
@@ -23,25 +37,26 @@ def test_run(self):
         fetch of data.  For instance if you have a limit of 250 records ensure
         that 251 (or more) records have been posted for that stream.
         """
-        conn_id = self.create_connection()
 
         # Select all streams and all fields within streams
         found_catalogs = menagerie.get_catalogs(conn_id)
         incremental_streams = {key for key, value in self.expected_replication_method().items()
-                               if value == self.INCREMENTAL}
+                               if value == self.INCREMENTAL and key in testable_streams}
+
 
-        untested_streams = self.child_streams().union({'abandoned_checkouts', 'collects', 'metafields', 'transactions', 'order_refunds'})
+        # our_catalogs = [catalog for catalog in found_catalogs if
+        #                 catalog.get('tap_stream_id') in incremental_streams.difference(
+        #                     untested_streams)]
         our_catalogs = [catalog for catalog in found_catalogs if
-                        catalog.get('tap_stream_id') in incremental_streams.difference(
-                            untested_streams)]
+                        catalog.get('tap_stream_id') in testable_streams]
 
 
         self.select_all_streams_and_fields(conn_id, our_catalogs, select_all_fields=True)
         # Run a sync job using orchestrator
         record_count_by_stream = self.run_sync(conn_id)
         actual_fields_by_stream = runner.examine_target_output_for_fields()
 
-        for stream in self.expected_streams().difference(untested_streams):
+        for stream in testable_streams:
             with self.subTest(stream=stream):
 
                 # verify that we can paginate with all fields selected
```

Comment:
I don't think the expected value should ever fall back to zero. It would only cover up the fact that we did not get past the page limit. Also shouldn't the limit be set to the `results_per_page` from get_properties since we are overwriting the default?

Reply:
> Also shouldn't the limit be set to the results_per_page from get_properties since we are overwriting the default?

I agree with this and can make the change and test it